pkg/logproto/compat.go

```diff
@@ -367,24 +367,13 @@ func (m *FilterChunkRefRequest) GetQuery() string {
 		chunksHash = h.Sum64()
 	}
 
-	// Short circuit if there are no filters.
-	if len(m.Filters) == 0 {
+	plan := m.Plan.String()
+	if plan == "" {
 		return fmt.Sprintf("%d", chunksHash)
 	}
 
-	var sb strings.Builder
-	for i, filter := range m.Filters {
-		if i > 0 {
-			sb.WriteString(",")
-		}
-		sb.Write(fmt.Appendf(encodeBuf[:0], "%d", filter.Ty))
-		sb.WriteString("-")
-		sb.WriteString(filter.Match)
-		sb.WriteString("-")
-		sb.WriteString(filter.Op)
-	}
-
-	return fmt.Sprintf("%d/%s", chunksHash, sb.String())
+	// TODO(salvacorts): plan.String() will return the whole query. This is not optimal since we are only interested in the filter expressions.
+	return fmt.Sprintf("%d/%s", chunksHash, plan)
```

Comment:
Do we have a limit of the cache key length? Plan could be up to 2k characters, right?
Would a query hash be sufficient, since we have the combination with the chunksHash?

Reply:
Done, changed to use plan hash